qutip/tests/test_mesolve.py

```diff
@@ -920,5 +920,53 @@ def f(t, args):
             msg="evolution with feedback not proceding as expected")
 
 
+def test_non_hermitian_dm():
+    """Test that mesolve works correctly for density matrices that are
+    not Hermitian.
+    See Issue #1460
+    """
+    N = 2
+    a = destroy(N)
+    n = num(N)
+    x = (a + a.dag())/np.sqrt(2)
+    H = a.dag() * a
+
+    rho0 = x*fock_dm(N, 0)
+    tlist = np.linspace(0, 0.1, 2)
+
+    options = Options()
+    options.store_final_state = True
+    options.store_states = True
+
+    result = mesolve(H, rho0, tlist, e_ops=[x], options=options)
+
+    imag_part = np.abs(np.imag(result.expect[0][-1]))
+    # Since we used an initial state that is not Hermitian, the expectation of
+    # x must be imaginary for t>0.
+    msg = ('Mesolve is not working properly with a non Hermitian density ' +
+           ' matrix as input. Check computation of '
+          )
```

Comment:
Just for the future: these messages aren't actually necessary, though I get that you were just copying the style of the rest of the file (which _is_ a good thing to do in general!).

A failed assertion already means that something isn't working properly, and `pytest` will tell use exactly which assertion failed and show the code context, so the strings don't actually help too much.

Reply:
Thank you for the comment! I will take that into account in the future when writing tests.